webapp/src/shared/types.ts

```diff
@@ -7,3 +7,35 @@ export enum Color {
   BACKGROUND = 'text-color-background',
   DARK = 'text-color-dark',
 }
+
+export interface User {
+  readonly username: string;
+  readonly email: string;
+  readonly avatarId: string | null;
+  readonly id: string;
+  readonly createdAt: Date;
+} /* This could be autogenerated - it's provisionally here, since it's a shared type */
+
+export function instanceOfUser(value: object): boolean {
```

Comment:
thought: Since we are the ones developing the back-end, maybe we could trust ourselves to return valid HTTP status codes and response types (or maybe not? :sweat_smile:)

sugestion: We could check if the response has a valid status code ([Response.ok](https://developer.mozilla.org/en-US/docs/Web/API/Response/ok)) and simplify the error checking
https://developer.mozilla.org/en-US/docs/Learn/JavaScript/Client-side_web_APIs/Fetching_data#the_can_store

Reply:
What do we even need the openApi generator for, then?

Reply:
I think we should use it, even if we are the ones developing the backend. Not because we don't trust ourselves, but because it's good practice.

Reply:
Ok, good point :+1: